File: backend/app/api/comments.py
from fastapi import APIRouter, Depends, HTTPException, Query
from sqlalchemy.orm import Session
from app.database.session import get_db
from app.auth.dependencies import get_current_user
from app.models.user import User
from app.schemas.comment import CommentResponse, CommentReply
from app.services.comment_service import CommentService

router = APIRouter(
    prefix="/comments",
    tags=["Comments"]
)

@router.post("/accounts/{account_id}/sync", response_model=list[CommentResponse])
async def sync_comments(
    account_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Sync and fetch latest comments for a connected social media account.
    """
    return await CommentService.sync_comments(db, current_user.id, account_id)

@router.get("/accounts/{account_id}", response_model=list[CommentResponse])
async def get_comments(
    account_id: int,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Get all comments for an account.
    """
    return await CommentService.get_comments(db, current_user.id, account_id)

@router.post("/accounts/{account_id}/{comment_id}/reply", response_model=CommentResponse)
async def reply_to_comment(
    account_id: int,
    comment_id: str,
    payload: CommentReply,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Post a reply to a comment.
    """
    return await CommentService.reply_to_comment(db, current_user.id, account_id, comment_id, payload.content)

@router.post("/accounts/{account_id}/{comment_id}/hide", response_model=CommentResponse)
async def hide_comment(
    account_id: int,
    comment_id: str,
    hide: bool = Query(True),
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Hide or unhide a comment.
    """
    return await CommentService.hide_comment(db, current_user.id, account_id, comment_id, hide)

@router.delete("/accounts/{account_id}/{comment_id}")
async def delete_comment(
    account_id: int,
    comment_id: str,
    db: Session = Depends(get_db),
    current_user: User = Depends(get_current_user)
):
    """
    Delete a comment.
    """
    await CommentService.delete_comment(db, current_user.id, account_id, comment_id)
    return {"message": "Comment deleted successfully"}

File: frontend/services/comments.ts
import apiClient from "./api";

export interface Comment {
  id: number;
  user_id: number;
  account_id: number;
  post_id?: number;
  platform_post_id: string;
  platform_comment_id: string;
  parent_id?: string;
  sender_id: string;
  sender_name?: string;
  content: string;
  is_hidden: boolean;
  is_deleted: boolean;
  is_from_me: boolean;
  sent_at: string;
  created_at: string;
}

export const commentsService = {
  async syncComments(accountId: number): Promise<Comment[]> {
    const response = await apiClient.post<Comment[]>(`/comments/accounts/${accountId}/sync`);
    return response.data;
  },

  async getComments(accountId: number): Promise<Comment[]> {
    const response = await apiClient.get<Comment[]>(`/comments/accounts/${accountId}`);
    return response.data;
  },

  async replyToComment(accountId: number, commentId: string, content: string): Promise<Comment> {
    const response = await apiClient.post<Comment>(`/comments/accounts/${accountId}/${commentId}/reply`, {
      content,
    });
    return response.data;
  },

  async hideComment(accountId: number, commentId: string, hide: boolean): Promise<Comment> {
    const response = await apiClient.post<Comment>(`/comments/accounts/${accountId}/${commentId}/hide`, null, {
      params: { hide },
    });
    return response.data;
  },

  async deleteComment(accountId: number, commentId: string): Promise<void> {
    await apiClient.delete(`/comments/accounts/${accountId}/${commentId}`);
  },
};
